packages/di/src/providers.ts

```diff
@@ -0,0 +1,97 @@
+import { readInjectableMetadata } from './metadata';
+
+export const Type = Function;
```

Comment:
Why `Function`? Maybe this could be more type-safe somehow?

Reply:
Function because type can be a class or function, it doesn't matter really for type safety if we define the Type as an object with constructor and Function because Function is an object with constructor pretty much.